File: Server/src/main/content/region/misthalin/quest/free/shieldofarrav/plugin/CertificatePlugin.java
package content.region.misthalin.quest.free.shieldofarrav.plugin;

import core.game.interaction.NodeUsageEvent;
import core.game.interaction.UseWithHandler;
import core.game.node.entity.player.Player;
import core.game.node.item.Item;
import core.plugin.Initializable;
import core.plugin.Plugin;

/**
 * The Certificate plugin.
 */
@Initializable
public final class CertificatePlugin extends UseWithHandler {

    private static final Item CERTIFICATE = new Item(769);

    /**
     * Instantiates a new Certificate plugin.
     */
    public CertificatePlugin() {
        super(11173);
    }

    @Override
    public Plugin<Object> newInstance(Object arg) throws Throwable {
        addHandler(11174, ITEM_TYPE, this);
        return this;
    }

    @Override
    public boolean handle(NodeUsageEvent event) {
        final Player player = event.getPlayer();
        if (player.getInventory().remove(new Item(event.getUsedItem().getId(), 1), new Item(event.getBaseItem().getId(), 1))) {
            player.getInventory().add(CERTIFICATE);
        }
        return true;
    }

}

File: Emulator/src/main/content/region/misthalin/varrock/quest/phoenixgang/handlers/CertificatePlugin.kt
package content.region.misthalin.varrock.quest.phoenixgang.handlers

import core.game.interaction.NodeUsageEvent
import core.game.interaction.UseWithHandler
import core.game.node.item.Item
import core.plugin.Initializable
import core.plugin.Plugin

/**
 * Represents the Certificate plugin.
 */
@Initializable
class CertificatePlugin : UseWithHandler(11173) {

    override fun newInstance(arg: Any?): Plugin<Any> {
        addHandler(11174, ITEM_TYPE, this)
        return this
    }

    override fun handle(event: NodeUsageEvent): Boolean {
        val player = event.player
        if (player.inventory.remove(Item(event.usedItem.id, 1), Item(event.baseItem.id, 1))) {
            player.inventory.add(CERTIFICATE)
        }
        return true
    }

    companion object {
        private val CERTIFICATE = Item(769)
    }
}
